src/segment_git.go

```diff
@@ -389,25 +400,35 @@ func (g *git) hasGitFile(file string) bool {
 	return g.env.hasFilesInDir(g.repo.gitWorkingFolder, file)
 }
 
-func (g *git) hasGitFolder(folder string) bool {
-	path := g.repo.gitWorkingFolder + "/" + folder
-	return g.env.hasFolder(path)
+func (g *git) hasGitFolder(folder string, rootFolder bool) bool {
```

Comment:
Still feels off. I see what you meant with avoiding to add that context to every call. This way we're better off replacing the call with the actual `env` call to avoid that boolean and simply fall back to the relevant call and context in line. Thoughts?

Reply:
I tried a move to that direction. Direct access to `env` for `hasGitFolder` and folder parameter added to `getGitFileContents`.  
Tell me what you think.